src/View.Uno.Shared/Controls/SwipeRefresh/SwipeRefresh.WinUI.cs

```diff
@@ -0,0 +1,127 @@
+﻿#if WINDOWS_WINUI
+using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Nventive.View.Controls
+{
+	[TemplatePart(Name = ScrollViewerPartName, Type = typeof(ScrollViewer))]
+	[TemplatePart(Name = SnapPanelPartName, Type = typeof(SnapPanel))]
+	[TemplatePart(Name = RefreshContainerPartName, Type = typeof(FrameworkElement))]
+	[TemplateVisualState(GroupName = SwipeVisualStates, Name = ReleasedVisualState)]
+	[TemplateVisualState(GroupName = SwipeVisualStates, Name = SwipingVisualState)]
+	[TemplateVisualState(GroupName = SwipeVisualStates, Name = SwipedVisualState)]
+	public partial class SwipeRefresh
+	{
+		private const string SnapPanelPartName = "PART_SnapPanel";
+		private const string ScrollViewerPartName = "PART_ScrollViewer";
+		private const string RefreshContainerPartName = "PART_RefreshContainer";
+
+		private const string SwipeVisualStates = "Swipe";
+		private const string SwipingVisualState = "Swiping";
+		private const string SwipedVisualState = "Swiped";
+		private const string ReleasedVisualState = "Released";
+
+		private ScrollViewer _scrollViewer;
+		private SnapPanel _snapPanel;
+		private FrameworkElement _refreshContainer;
+
+		public SwipeRefresh()
+		{
+			DefaultStyleKey = typeof(SwipeRefresh);
+		}
+
+		/// <summary>
+		/// Gets or sets the offset before considering the content swiped. The default value is 100.
+		///
+		/// </summary>
+		/// <remarks>The view cannot bind to this property. It must be set once.</remarks>
+		public double SwipedSize { get; set; } = 100D;
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+
+			_scrollViewer = this.GetTemplateChild(ScrollViewerPartName) as ScrollViewer;
+			if (_scrollViewer == null)
+			{
+				throw new NotSupportedException("SwipeRefresh requires a ScrollViewer element named " + ScrollViewerPartName + ".");
+			}
+			_snapPanel = this.GetTemplateChild(SnapPanelPartName) as SnapPanel;
+			if (_scrollViewer == null)
+			{
+				throw new NotSupportedException("SwipeRefresh requires a SnapPanel element named " + SnapPanelPartName + ".");
+			}
+			_refreshContainer = this.GetTemplateChild(RefreshContainerPartName) as FrameworkElement;
+			if (_scrollViewer == null)
+			{
+				throw new NotSupportedException("SwipeRefresh requires a container named " + RefreshContainerPartName + ".");
+			}
+
+			_scrollViewer.ViewChanging += OnScrollViewerViewChanging;
```

Comment:
we should definitly unsubscribe there first in case this method is called again (can happen on iOS)